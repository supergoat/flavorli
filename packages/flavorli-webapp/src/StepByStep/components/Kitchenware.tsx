import React from 'react';
import {Stack, Text, H3} from '@flavorli/elements';
import {IKitchenware} from '../types';

const Kitchenware = ({kitchenware}: {kitchenware: IKitchenware[]}) => {
  return !!kitchenware.length ? (
    <Stack gap={8} width="100%">
      <H3>You will need</H3>

      {kitchenware.map(item => (
        <Text key={item.name}>
          {item.qty} {item.name}
        </Text>
      ))}
    </Stack>
  ) : null;
};

export default Kitchenware;
