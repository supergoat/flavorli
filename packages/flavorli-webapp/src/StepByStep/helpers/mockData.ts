import {IStep} from '../types';

export const steps: IStep[] = [
  {
    no: 1,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description: 'Place a pot of water in medium heat and set aside to boil',
    ingredients: [],
    kitchenware: [
      {
        qty: '1',
        name: 'Pot',
      },
    ],
  },
  {
    no: 2,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description: 'Place a sauce pan over high heat and add the olive oil',
    ingredients: [
      {
        qty: '4 - 6 tbsps',
        name: 'olive oil',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Sauce Pan',
      },
    ],
  },
  {
    no: 3,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Cut the eggplan into 1 cm cubes. Add it to the pan and sauté for 1-2 minutes',
    ingredients: [
      {
        qty: '2',
        name: 'eggplants',
      },
    ],
    kitchenware: [
      {qty: '1', name: 'Knife'},
      {qty: '1', name: 'Cutting Board'},
      {qty: '1', name: 'Spoon'},
    ],
  },
  {
    no: 4,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Finely chop the onion, add it to the pan and sauté for 1-2 minutes',
    ingredients: [
      {
        qty: '1',
        name: 'onion',
      },
    ],
    kitchenware: [],
  },
  {
    no: 5,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Cut the zucchinis and the peppers into small cubes and add them to the pan. Sauté for 4-5 minutes, until the vegetables are ready',
    ingredients: [
      {
        qty: '2',
        name: 'zucchinis',
      },
      {
        qty: '1',
        name: 'red bell pepper',
      },
      {
        qty: '1',
        name: 'green bell pepper',
      },
    ],
    kitchenware: [],
  },
  {
    no: 6,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Add the salt, pepper, oregano, thyme, chili flakes, sugar and tomato paste and sauté',
    ingredients: [
      {
        qty: '',
        name: 'salt',
      },
      {
        qty: '',
        name: 'pepper',
      },
      {
        qty: '1 tsp',
        name: 'oregano',
      },
      {
        qty: '1 tbsp',
        name: 'thyme',
      },
      {
        qty: '1 tsp',
        name: 'chilli flakes',
      },
      {
        qty: '1 tbsp',
        name: 'granulated sugar',
      },
      {
        qty: '1 tsp',
        name: 'tomato paste',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Tea Spoon',
      },
    ],
  },
  {
    no: 7,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Add the wine and canned tomatoes and simmer at medium-low heat for 10 minutes',
    ingredients: [
      {
        qty: '40 g',
        name: 'white wine',
      },
      {
        qty: '400 g',
        name: 'canned tomatoes',
      },
    ],
    timer: {id: 7, name: 'Step 7: Simmer Sauce', minutes: 10, seconds: 0},
    kitchenware: [{qty: '1', name: 'scale'}],
  },
  {
    no: 8,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [{from: 1, heading: 'From Step 1', name: 'Boiling water'}],
    description:
      'In the pot with boiling water, add salt along with the macaroni, and boil for ten minutes',
    ingredients: [
      {
        qty: '',
        name: 'salt',
      },
      {
        qty: '500 g',
        name: 'macaroni',
      },
    ],
    timer: {id: 8, name: 'Step 8: Cook Pasta', minutes: 10, seconds: 0},
    kitchenware: [],
  },
  {
    no: 9,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [],
    description: 'Preheat the oven at 180 C set to fan',
    ingredients: [],
    kitchenware: [],
  },
  {
    no: 10,
    tag: {
      text: 'Bechamel',
      color: 'tagOrange',
    },
    links: [],
    description:
      'Place a pot over medium heat. Add the olive oil and the flour and using a hand whisk, mix well until all of the flour is absorbed',
    ingredients: [
      {
        qty: '80 g',
        name: 'olive oil',
      },
      {
        qty: '80 g',
        name: 'all-purpose flour',
      },
    ],
    kitchenware: [
      {qty: '1', name: 'Small Pot'},
      {qty: '1', name: 'Hand Whisk'},
    ],
  },
  {
    no: 11,
    tag: {
      text: 'Bechamel',
      color: 'tagOrange',
    },
    links: [],
    description: 'Add the soy milk in 7-8 batches, stirring constantly',
    ingredients: [
      {
        qty: '800 g',
        name: 'soy milk',
      },
    ],
    kitchenware: [],
  },
  {
    no: 12,
    tag: {
      text: 'Bechamel',
      color: 'tagOrange',
    },
    links: [],
    description:
      'Season with salt and pepper and as soon as it thickens and comes to boil, remove from the heat',
    ingredients: [
      {
        qty: '',
        name: 'salt',
      },
      {
        qty: '',
        name: 'pepper',
      },
    ],
    kitchenware: [],
  },
  {
    no: 13,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [
      {
        from: 7,
        heading: 'From Step 7',
        name: 'Pan with vegetables',
        timerId: 7,
      },
      {
        from: 8,
        heading: 'From Step 8',
        name: 'Macaroni',
        timerId: 8,
      },
    ],
    description:
      'Drain the macaroni and transfer to the pan with the vegetables. Mix and set aside',
    ingredients: [],
    kitchenware: [{qty: '1', name: 'Colander'}],
  },
  {
    no: 14,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [{from: 12, heading: 'From Step 12', name: 'Bechamel sauce'}],
    description:
      'Add 100g of the bechamel into the pasta mixture, add oregano, mint, and mix well',
    ingredients: [
      {
        qty: '1 tsp',
        name: 'oregano',
      },
      {
        qty: '1 tbsp',
        name: 'mint',
      },
    ],
    kitchenware: [],
  },
  {
    no: 15,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [],
    description:
      'Grease 40 cm ovenproof baking dish with olive oil and sprinke the breadcrumps',
    ingredients: [
      {
        qty: '1 tbsp',
        name: 'olive oil',
      },
      {
        qty: '25 g',
        name: 'breadcrumps',
      },
    ],
    kitchenware: [{qty: '1', name: '40 cm ovenproof baking dish'}],
  },
  {
    no: 16,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [],
    description:
      'Add the pasta mixture, spread the bechamel sauce, and sprinke the breadcrumps, olive oil, thyme and pepper',
    ingredients: [
      {
        qty: '25 g',
        name: 'breadcrumps',
      },
      {
        qty: '1 tbsp',
        name: 'olive oil',
      },
      {
        qty: '1 tbsp',
        name: 'thyme',
      },
      {
        qty: '',
        name: 'pepper',
      },
    ],
    kitchenware: [],
  },
  {
    no: 17,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [],
    description: 'Bake for 25 - 30 minutes. Remove and let it cool',
    ingredients: [],
    timer: {id: 17, name: 'Step 17: Bake Pastitsio', minutes: 30, seconds: 0},
    kitchenware: [
      {
        qty: '1',
        name: 'Oven Mitts',
      },
    ],
  },
];
