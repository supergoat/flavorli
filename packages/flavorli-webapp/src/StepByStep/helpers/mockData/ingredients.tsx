export const ingredients = [
  {
    qty: '2',
    name: 'eggplants',
  },
  {
    qty: '1',
    name: 'onion',
  },
  {
    qty: '1 clove',
    name: 'garlic',
  },
  {
    qty: '2',
    name: 'zucchinis',
  },
  {
    qty: '1',
    name: 'red bell pepper',
  },
  {
    qty: '1',
    name: 'green bell pepper',
  },
  {
    qty: '2 teaspoon',
    name: 'oregano',
  },
  {
    qty: '2 tablespoon',
    name: 'thyme',
  },
  {
    qty: '1 teaspoon',
    name: 'chilli flakes',
  },
  {
    qty: '1 tablespoon',
    name: 'granulated sugar',
  },
  {
    qty: '1 tablespoon',
    name: 'tomato paste',
  },
  {
    qty: '50 g',
    name: 'white wine',
  },
  {
    qty: '400 g',
    name: 'canned tomatoes',
  },
  {
    qty: '500 g',
    name: 'macaroni',
  },
  {
    qty: '80 g',
    name: 'all-purpose flour',
  },
  {
    qty: '80 g',
    name: 'olive oil',
  },
  {
    qty: '800 g',
    name: 'soy milk',
  },
  {
    qty: '50 g',
    name: 'dry breadcrumps',
  },
  {
    qty: '1 tablespoon',
    name: 'mint',
  },
  {
    qty: '1',
    name: 'squeeze bottle olive oil',
  },
  {
    qty: '',
    name: 'salt',
  },
  {
    qty: '',
    name: 'pepper',
  },
];
