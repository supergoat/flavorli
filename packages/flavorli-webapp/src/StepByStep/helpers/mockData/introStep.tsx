import {IIntro} from '../../types';

export const introStep: IIntro = {
  name: 'Greek Pastitsio',
  image:
    'https://akispetretzikis.com/system/uploads/medium/data/14955/recipe_main_vegan-pastitsio.jpg',
  preparation: '25 minutes',
  cooking: '40 minutes',
  portions: '8 - 10',
  difficulty: 'Level 2',
};
