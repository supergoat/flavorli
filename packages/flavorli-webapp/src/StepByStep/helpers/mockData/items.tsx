export const items = [
  {
    qty: '2',
    name: 'Bowls',
  },
  {
    qty: '1',
    name: 'Small Bowl',
  },
  {
    qty: '3',
    name: 'Glasses',
  },
  {
    qty: '11',
    name: 'Ramekins',
  },
  {
    qty: '1',
    name: 'Sauce Pan',
  },
  {
    qty: '1',
    name: 'Pot',
  },
  {
    qty: '1',
    name: 'Small Pot',
  },
  {
    qty: '1',
    name: 'Cutting Board',
  },
  {
    qty: '1',
    name: 'Scale',
  },
  {
    qty: '1',
    name: 'Hand Whisk',
  },
  {
    qty: '1',
    name: 'Colander',
  },
  {
    qty: '1',
    name: '40 cm ovenproof dish',
  },
];
