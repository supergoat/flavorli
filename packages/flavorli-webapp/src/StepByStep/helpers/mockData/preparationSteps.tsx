import {IPreparationStep} from '../../types';

export const preparationSteps: IPreparationStep[] = [
  {
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    description:
      'Add the oregano, thyme, chili flakes, sugar and tomato paste into ramekins and set aside',
    ingredients: [
      {
        qty: '1 teaspoon',
        name: 'oregano',
      },
      {
        qty: '1 tablespoon',
        name: 'thyme',
      },
      {
        qty: '1 teaspoon',
        name: 'chili flakes',
      },
      {
        qty: '1 tablespoon',
        name: 'granulated sugar',
      },
      {
        qty: '1 tablespoon',
        name: 'tomato paste',
      },
    ],
    kitchenware: [
      {
        qty: '5',
        name: 'Ramekins',
      },
    ],
    images: [],
  },
  {
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    description:
      'Measure the wine in a glass and set aside. Wash and wipe the can of tomatoes and set aside',
    ingredients: [
      {
        qty: '50 g',
        name: 'white wine',
      },
      {
        qty: '400 g',
        name: 'canned tomatoes',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Glass',
      },
    ],
    images: [],
  },
  {
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    description: 'Measure the macaroni and set aside in a bowl',
    ingredients: [
      {
        qty: '500 g',
        name: 'macaroni',
      },
    ],
    kitchenware: [],
    images: [],
  },
  {
    tag: {
      text: 'Bechamel',
      color: 'tagOrange',
    },
    description:
      'Measure the olive oil in a glass and set aside. Measure the flour in a glass and set aside.',
    ingredients: [
      {
        qty: '80 g',
        name: 'olive oil',
      },
      {
        qty: '80 g',
        name: 'all-purpose flour',
      },
    ],
    kitchenware: [{qty: '2', name: 'Glass'}],
    images: [],
  },
  {
    tag: {
      text: 'Bechamel',
      color: 'tagOrange',
    },
    description: 'Set a carton of soy milk aside',
    ingredients: [
      {
        qty: '800 g',
        name: 'soy milk',
      },
    ],
    kitchenware: [],
    images: [],
  },
  {
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    description: 'Add the oregano and mint into ramekins',
    ingredients: [
      {
        qty: '1 teaspoon',
        name: 'oregano',
      },
      {
        qty: '1 tablespoon',
        name: 'mint',
      },
    ],
    kitchenware: [
      {
        qty: '2',
        name: 'Ramekin',
      },
    ],
    images: [],
  },
  {
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    description: 'Add the breadcrumps in a glass',
    ingredients: [
      {
        qty: '50 g',
        name: 'breadcrumps',
      },
    ],
    kitchenware: [{qty: '1', name: 'Glass'}],
    images: [],
  },
  {
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    description: 'Add the thyme into a ramekin',
    ingredients: [
      {
        qty: '1 tablespoon',
        name: 'thyme',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Ramekin',
      },
    ],
    images: [],
  },
  {
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    description: 'Add the oregano and mint into ramekins',
    ingredients: [
      {
        qty: '1 oregano',
        name: 'thyme',
      },
      {
        qty: '1 tablespoon',
        name: 'mint',
      },
    ],
    kitchenware: [
      {
        qty: '2',
        name: 'Ramekins',
      },
    ],
    images: [],
  },
  {
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    description: 'Cut the eggplants into 1 cm cubes and place them in a bowl',
    ingredients: [
      {
        qty: '2',
        name: 'eggplants',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Bowl',
      },
    ],
    images: [
      {
        src: require('../../images/eggplant_1.png'),
        alt: 'Chef cutting the top of the eggplant off',
      },
      {
        src: require('../../images/eggplant_2.png'),
        alt:
          'Chef placing the eggplant vertically, with the cut top facing upward, cutting the eggplant into columns',
      },
      {
        src: require('../../images/eggplant_3.png'),
        alt:
          'Chef stacking half the eggplant columns horizontally and cutting the eggplant in columns',
      },
      {
        src: require('../../images/eggplant_4.png'),
        alt: '',
      },
      {
        src: require('../../images/eggplant_5.png'),
        alt:
          'Chef putting the eggplant together and cutting into little squares',
      },
    ],
  },
  {
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    description: 'Finely chop the onion and place it in a ramekin',
    ingredients: [
      {
        qty: '1',
        name: 'onion',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Ramekin',
      },
    ],
    images: [
      {
        src: require('../../images/onion_1.png'),
        alt: 'Cutting Onion 1',
      },
      {
        src: require('../../images/onion_2.png'),
        alt: 'Cutting Onion 2',
      },
      {
        src: require('../../images/onion_3.png'),
        alt: 'Cutting Onion 3',
      },
    ],
  },
  {
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    description: 'Finely chop the garlic and place it in a ramekin',
    ingredients: [
      {
        qty: '1',
        name: 'garlic',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Ramekin',
      },
    ],
    images: [
      {
        src: require('../../images/garlic.png'),
        alt: 'Cutting Garlic 1',
      },
    ],
  },
  {
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    description: 'Cut the zucchinis into small cubes and place them in a bowl',
    ingredients: [
      {
        qty: '2',
        name: 'zucchinis',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Bowl',
      },
    ],
    images: [
      {
        src: require('../../images/zucchini_1.png'),
        alt: 'Cutting Zucchini 1',
      },
      {
        src: require('../../images/zucchini_2.png'),
        alt: 'Cutting Zucchini  2',
      },
      {
        src: require('../../images/zucchini_3.png'),
        alt: 'Cutting Zucchini 3',
      },
      {
        src: require('../../images/zucchini_4.png'),
        alt: 'Cutting Zucchini 4',
      },
    ],
  },
  {
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    description:
      'Cut the the peppers into small cubes and place them in a small bowl',
    ingredients: [
      {
        qty: '1',
        name: 'red bell pepper',
      },
      {
        qty: '1',
        name: 'green bell pepper',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Small Bowl',
      },
    ],
    images: [
      {
        src: require('../../images/pepper.png'),
        alt: 'Cutting the peppers 1',
      },
    ],
  },
];
