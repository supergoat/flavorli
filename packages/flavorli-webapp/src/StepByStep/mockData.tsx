import {IStep, IRecipeTimer, ILink, IRecipe} from '../types';

export const recipes: IRecipe[] = [
  {
    id: '1',
    createdAt: '2020-01-19T19:37:42.873Z',
    author: 'Robin Robertson',
    name: 'Cherry Berry Smoothie',
    image:
      'https://flavorli-bucket-2.s3.eu-west-2.amazonaws.com/CherryBerrySmoothie.jpeg',
    preparationTime: 7,
    cookingTime: 0,
    portions: '1',
    difficulty: '1',
    ingredients: [
      {
        qty: '1 cup/100 g',
        name: 'Frozen Blueberries',
      },
      {
        qty: '1/2 cup/110 g',
        name: 'fresh or frozen pitted cherries',
      },
      {
        qty: '1',
        name: 'frozen ripe banana, cut into chunks before freezing',
      },
      {
        qty: '1 tbsp',
        name: 'ground flaxseeds (or linseeds)',
      },
      {
        qty: '1 tbsp',
        name: 'almond butter (or peanut butter)',
      },
      {
        qty: '1 cup/22.5 g',
        name: 'raw spinach leaves',
      },
      {
        qty: '1 1/2 cup (360 ml)',
        name: 'water',
      },
    ],
    items: [
      {
        qty: '1',
        name: 'Blender',
      },
      {
        qty: '1',
        name: 'Cup',
      },
      {
        qty: '1/2',
        name: 'cup',
      },
      {
        qty: '1',
        name: 'tablespoon',
      },
    ],
    steps: [
      {
        no: 1,
        type: 'MISE_EN_PLACE',
        tag: {
          text: 'Smoothie',
          color: 'tagRed',
        },
        items: [
          {
            qty: '1',
            name: 'Blender',
          },
        ],
        ingredients: [
          {
            qty: '1 cup/100 g',
            name: 'Frozen Blueberries',
          },
          {
            qty: '1/2 cup/110 g',
            name: 'fresh or frozen pitted cherries',
          },
          {
            qty: '1',
            name: 'frozen ripe banana, cut into chunks before freezing',
          },
          {
            qty: '1 tbsp',
            name: 'ground flaxseeds (or linseeds)',
          },
          {
            qty: '1 tbsp',
            name: 'almond butter (or peanut butter)',
          },
          {
            qty: '1 cup/22.5 g',
            name: 'raw spinach leaves',
          },
          {
            qty: '1 1/2 cup (360 ml)',
            name: 'water',
          },
        ],
        tasks: [
          'Add 1 cup/100 g Frozen Blueberries in a blender',
          'Add 1/2 cup/110 g fresh or frozen pitted cherries',
          'Add 1 frozen ripe banana, cut into chunks before freezing',
          'Add 1 tbsp ground flaxseeds (or linseeds)',
          'Add 1 tbsp almond butter (or peanut butter)',
          'Add 1 cup/22.5 g raw spinach leaves',
          'Add 1 1/2 cup cup (360 ml) water',
        ],
        images: [],
      },
      {
        no: 2,
        type: 'PREPARATION',
        tag: {
          text: 'Smoothie',
          color: 'tagRed',
        },
        items: [],
        ingredients: [],
        tasks: [
          'Blend all the ingredients until smooth and creamy for about 1 minute',
          'Add more water if you prefer a thinner consistency',
          'Serve immediately',
        ],
        timer: {
          id: '2',
          name: 'Step 2: Blend',
          minutes: 1,
          seconds: 0,
        },
        images: [],
      },
    ],
  },
];

export const timer: IRecipeTimer = {
  id: '1',
  name: 'Thicken Sauce',
  minutes: 10,
  seconds: 0,
  isPaused: true,
  updatedAt: '2020-01-26T15:15:06.689Z',
  remainingTime: 600000,
};

export const link: ILink = {
  from: 1,
  heading: 'From Step 1',
  name: 'Thicken Sauce',
};

export const linkWithTimer: ILink = {
  from: 1,
  heading: 'From Step 1',
  name: 'Thicken Sauce',
  timerId: '1',
};

export const items = [
  {
    qty: '1',
    name: 'Pot',
  },
];

export const preparationStep: IStep = {
  type: 'MISE_EN_PLACE',
  tag: {
    text: 'Sauce',
    color: 'tagRed',
  },
  tasks: ['Add the coconul oil in a ramekin and place it first in order'],
  ingredients: [
    {
      qty: '2 tbsps',
      name: 'coconut oil',
    },
  ],
  items: [
    {
      qty: '1',
      name: 'Ramekin',
    },
  ],
  images: [],
};

export const recipeStep: IStep = {
  type: 'PREPARATION',
  no: 1,
  tag: {
    text: 'Sauce',
    color: 'tagRed',
  },
  links: [],
  tasks: ['Heat the coconut oil in a small pot on low heat'],
  ingredients: [
    {
      qty: '2 tbsps',
      name: 'coconut oil',
    },
  ],
  items: [
    {
      qty: '1',
      name: 'Small Pot',
    },
  ],
  images: [],
};

export const recipeSteps = [recipeStep];

export const introStep = {
  author: 'Akis Petratzikis',
  name: 'Greek Pastitsio',
  image: '',
  preparation: '25 minutes',
  cooking: '40 minutes',
  portions: '8 - 10',
  difficulty: 'Level 2',
};

export const ingredients = [{qty: '1', name: 'Onion'}];

export const images = [
  {
    src: '',
    alt: 'image 1',
  },

  {
    src: '',
    alt: 'image 2',
  },
];
