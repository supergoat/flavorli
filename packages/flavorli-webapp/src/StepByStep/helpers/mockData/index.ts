// https://www.youtube.com/watch?v=q1q3HfVvzQ0
import {ingredients} from './ingredients';
import {items} from './items';
import {preparationSteps} from './preparationSteps';
import {recipeSteps} from './recipeSteps';
import {IStepByStep} from '../../types';

export const stepByStep: IStepByStep = {
  intro: {
    name: 'Greek Pastitsio',
    image:
      'https://akispetretzikis.com/system/uploads/medium/data/14955/recipe_main_vegan-pastitsio.jpg',
    preparation: '25 minutes',
    cooking: '40 minutes',
    portions: '8 - 10',
    difficulty: 'Level 2',
  },
  ingredients,
  items,
  preparationSteps,
  recipeSteps,
};

export {ingredients, items, preparationSteps, recipeSteps};
