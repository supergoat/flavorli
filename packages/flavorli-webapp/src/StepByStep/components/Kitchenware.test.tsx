import React from 'react';
import {render} from '../../helpers/test-helpers';
import {axe} from 'jest-axe';
import Kitchenware from './Kitchenware';
import {steps} from '../helpers/mockData';

describe('Kitchenware', () => {
  it('should not have any axe violations', async () => {
    const {container} = render(
      <Kitchenware kitchenware={steps[0].kitchenware} />,
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should render correctly', () => {
    const {container} = render(
      <Kitchenware kitchenware={steps[0].kitchenware} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an empty div if the kitchenware are an empty array', () => {
    const {container} = render(<Kitchenware kitchenware={[]} />);
    expect(container.firstChild).toBeNull();
  });
});
