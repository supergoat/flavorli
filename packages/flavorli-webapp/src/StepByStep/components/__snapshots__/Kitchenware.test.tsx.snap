// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kitchenware should render correctly 1`] = `
<div
  class="sc-AykKH khQdtk"
  direction="vertical"
  width="100%"
>
  <h3
    class="sc-AykKF fqdNMu"
    color="primary"
  >
    You will need
  </h3>
  <p
    class="sc-AykKG fCXClx"
  >
    1
     
    Pot
  </p>
</div>
`;
